src/fprime_gds/common/pipeline/dictionaries.py

```diff
@@ -76,6 +79,15 @@ def load_dictionaries(self, dictionary, packet_spec, packet_set_name):
             )
             self._channel_name_dict = json_channel_loader.get_name_dict(None)
             self._channel_id_dict = json_channel_loader.get_id_dict(None)
+            # Fw Types
+            fw_types_loader = fprime_gds.common.loaders.fw_type_json_loader.FwTypeJsonLoader(
+                dictionary
+            )
+            self._fw_type_name_dict = fw_types_loader.get_name_dict(None)
+            if config:
+                # Update config to use Fw types defined in the JSON dictionary
+                for fw_type_name, fw_type in self._fw_type_name_dict.items():
```

Comment:
We're passing in a `config` object to another class' method in order to modify the first object. That makes it a little hard to follow, and we end up calling `load_dictionaries(...config=None)` multiple times.

Thoughts on refactoring this a little so that config doesn't get modified here, but rather for example in StandardPipeline where it is needed?

Reply:
Agreed!